crates/ruff_python_ast/src/identifier.rs

```diff
@@ -251,10 +228,13 @@ impl<'a> IdentifierTokenizer<'a> {
         while let Some(c) = self.cursor.bump() {
             match c {
                 c if is_python_identifier_start(c) => {
-                    let start = self.offset.add(self.cursor.offset()).sub(c.text_len());
+                    self.offset += self.cursor.token_len();
```

Comment:
You can avoid some of the offset math by calling cursor.start_token before the match (and I think cursor also gives you the offset, you can store that in a stack variable)

Reply:
IDK this part is pretty confusing. I think I'd have to call `start_token` before every bump, and then handle adding offsets in every branch of the match here? (cursor doesn't seem to expose an offset. My old implementation did, which simplified this specific usage a little, but I didn't feel strongly about preserving.)

Reply:
Ah yeah sorry. You would need to call start before every bump.

Reply:
Apology... accepted!